src/ngraph/op/gather_nd.cpp

```diff
@@ -0,0 +1,84 @@
+//*****************************************************************************
+// Copyright 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/gather_nd.hpp"
+#include "ngraph/shape.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+static int PARAMS = 0;
+static int INDICES = 1;
+
+shared_ptr<Node> op::GatherND::copy_with_new_args(const NodeVector& new_args) const
+{
+    check_new_args_count(this, new_args);
+    return make_shared<GatherND>(new_args.at(PARAMS), new_args.at(INDICES));
+}
+
+void op::GatherND::validate_and_infer_types()
+{
+    element::Type result_et = get_input_element_type(PARAMS);
+
+    const PartialShape& params_shape = get_input_partial_shape(PARAMS);
+    const PartialShape& indices_shape = get_input_partial_shape(INDICES);
+
+    NODE_VALIDATION_CHECK(this,
+                          indices_shape.rank().is_dynamic() ||
+                              static_cast<size_t>(indices_shape.rank()) >= 2,
+                          "indices rank is expected to be at least 2");
+
```

Comment:
why the rank is limited to at least 2? tf.gather_nd supports indices of rank size of 1. Is the plan to use gather op instead if the rank size is 1?